src/components/MessagesList/MessagesGroup/Message/Message.spec.js

```diff
@@ -414,6 +424,11 @@ describe('Message.vue', () => {
 						type: 'file',
 					},
 				}
+
+				useMessageInfoSpy.mockReturnValue({
```

Comment:
I prefer not to rely on passed props of the component as it may miss one of the needed keys like `timestamp`. 

> get rid of v-bind and set all props to Message/MessageBody/e.t.c explicitly (as a whole object) - could be done separately

If composable has a way to get it without extending component props, it is good as-is ? 

and regarding testing, we can move mocking level to store getter instead

Reply:
With the recent changed (order of v-bind) it will overwrite passed token in Vue 3.
 Ref: https://v3-migration.vuejs.org/breaking-changes/v-bind
 So it would be better to replace `<Message :token="token" v-bind="message" />` to `<Message :token="token" :message="message" />` and so on, but just for Next RC before migration to Vue 3 (as was discussed with @ShGKme)

Composable itself is good, but in Vue3 we'll face the issue in federated conversations, as `props.token` would be invalid for the local instance

We shouldn't mock our own modules, if they provide us logic, which we're testing. It's better to add complete message object to the store instead, if needed (`addMessage` or `processMessage`)

Reply:
> With the recent changed (order of v-bind) it will overwrite passed token in Vue 3.
Ref: [v3-migration.vuejs.org/breaking-changes/v-bind](https://v3-migration.vuejs.org/breaking-changes/v-bind)
So it would be better to replace <Message :token="token" v-bind="message" /> to <Message :token="token" :message="message" />

it turns out that token is now the same after [this](https://github.com/nextcloud/spreed/commit/ce6b67d9214f21c55119f434887ff06c3685b827) so both orders are the same. 

>  We should be able to refactor composables anytime, create or delete composables without the need to rewrite all the tests. It makes further refactoring very painful and doesn't help to test the component, it's only testing internal implementation.

That's why I mocked the composable output in the test instead of adding it to the store ( so composable can fetch from the store). 
If I understood correctly, composables are to be mocked in the UI test as we don't need their implementation testing, rather their output.

Reply:
Composables are a part of a component, not its interface.

It isn't important for a component's user if they are implemented via reusable composables or not. Tests should test how a component is used by other components and apps, not it is implemented internally.

This component renders a message using params from props and data from global state. Then we should test it with a specific prop in a specific global state as an integration test.

Currently, if we refactor it again, for example, remove, change or create a new composable, or refactor composable, we would need to rewrite component tests again.

If we break the composable, the test will pass even if it actually doesn't work (at least without unit tests on the composable).

In these cases, tests create more problems than they solve. They don't prevent all the issues, but they require doing an additional job on internal changes.

Reply:
`store.dispatch('processMessage', { token: TOKEN, message: messageProps })` works for that file without mocking, if you put it inside `function renderRichObject`